src/algebra/star/star_alg_hom.lean

```diff
@@ -0,0 +1,445 @@
+instance : has_coe_to_fun (A →⋆ₙₐ[R] B) (λ _, A → B) := fun_like.has_coe_to_fun
+
+initialize_simps_projections non_unital_star_alg_hom (to_fun → apply)
+
+@[simp] lemma coe_to_non_unital_alg_hom {f : A →⋆ₙₐ[R] B} :
+  (f.to_non_unital_alg_hom : A → B) = f := rfl
+
+@[ext] lemma ext {f g : A →⋆ₙₐ[R] B} (h : ∀ x, f x = g x) : f = g := fun_like.ext _ _ h
+
+/-- Copy of a `non_unital_star_alg_hom` with a new `to_fun` equal to the old one. Useful
+to fix definitional equalities. -/
+protected def copy (f : A →⋆ₙₐ[R] B) (f' : A → B) (h : f' = f) : A →⋆ₙₐ[R] B :=
+{ to_fun := f',
+  map_smul' := h.symm ▸ map_smul f,
+  map_zero' := h.symm ▸ map_zero f,
+  map_add' := h.symm ▸ map_add f,
+  map_mul' := h.symm ▸ map_mul f,
+  map_star' := h.symm ▸ map_star f }
+
+@[simp] lemma coe_mk (f : A → B) (h₁ h₂ h₃ h₄ h₅) :
+  ((⟨f, h₁, h₂, h₃, h₄, h₅⟩ : A →⋆ₙₐ[R] B) : A → B) = f :=
+rfl
+
+@[simp] lemma mk_coe (f : A →⋆ₙₐ[R] B) (h₁ h₂ h₃ h₄ h₅) :
+  (⟨f, h₁, h₂, h₃, h₄, h₅⟩ : A →⋆ₙₐ[R] B) = f :=
+by { ext, refl, }
+
+section
+variables (R A)
+/-- The identity as a cocompact continuous map. -/
+protected def id : A →⋆ₙₐ[R] A :=
+{ map_star' := λ x, rfl, .. (1 : A →ₙₐ[R] A) }
+
+@[simp] lemma coe_id : ⇑(non_unital_star_alg_hom.id R A) = id := rfl
+end
+
+/-- The composition of non-unital ⋆-algebra homomorphisms, as a non-unital ⋆-algebra
+homomorphism. -/
+def comp (f : B →⋆ₙₐ[R] C) (g : A →⋆ₙₐ[R] B) : A →⋆ₙₐ[R] C :=
+{ map_star' := by simp only [map_star, non_unital_alg_hom.to_fun_eq_coe, eq_self_iff_true,
+    non_unital_alg_hom.coe_comp, coe_to_non_unital_alg_hom, function.comp_app, forall_const],
+  .. f.to_non_unital_alg_hom.comp g.to_non_unital_alg_hom }
+
+@[simp] lemma coe_comp (f : B →⋆ₙₐ[R] C) (g : A →⋆ₙₐ[R] B) : ⇑(comp f g) = f ∘ g := rfl
+
+@[simp] lemma comp_apply (f : B →⋆ₙₐ[R] C) (g : A →⋆ₙₐ[R] B) (a : A) : comp f g a = f (g a) := rfl
+
+@[simp] lemma comp_assoc (f : C →⋆ₙₐ[R] D) (g : B →⋆ₙₐ[R] C) (h : A →⋆ₙₐ[R] B) :
+  (f.comp g).comp h = f.comp (g.comp h) := rfl
+
+@[simp] lemma id_comp (f : A →⋆ₙₐ[R] B) : (non_unital_star_alg_hom.id _ _).comp f = f :=
+ext $ λ _, rfl
+
+@[simp] lemma comp_id (f : A →⋆ₙₐ[R] B) : f.comp (non_unital_star_alg_hom.id _ _) = f :=
+ext $ λ _, rfl
+
+instance : monoid (A →⋆ₙₐ[R] A) :=
+{ mul := comp,
+  mul_assoc := comp_assoc,
+  one := non_unital_star_alg_hom.id R A,
+  one_mul := id_comp,
+  mul_one := comp_id, }
+
+@[simp] lemma coe_one : ((1 : A →⋆ₙₐ[R] A) : A → A) = id := rfl
+lemma one_apply (a : A) : (1 : A →⋆ₙₐ[R] A) a = a := rfl
+
+end basic
+
+section zero
+-- the `zero` requires extra type class assumptions because we need `star_zero`
+variables {R A B C D : Type*} [monoid R]
+variables [non_unital_non_assoc_semiring A] [distrib_mul_action R A] [star_add_monoid A]
+variables [non_unital_non_assoc_semiring B] [distrib_mul_action R B] [star_add_monoid B]
+
+instance : has_zero (A →⋆ₙₐ[R] B) :=
+⟨{ map_star' := by simp, .. (0 : non_unital_alg_hom R A B) }⟩
+
+instance : inhabited (A →⋆ₙₐ[R] B) := ⟨0⟩
```

Comment:
```suggestion
instance : inhabited (A →⋆ₙₐ[R] B) := ⟨0⟩
```
Can't we use the identity instead, and have this instance in the previous section?

Reply:
Yeah, we could, but that's only good for `A →⋆ₙₐ[R] A`, not morphisms from `A` to `B`. Which is preferable?

Reply:
Oh, right -- never mind, might as well leave it like this.